js/trbinance.js

```diff
@@ -0,0 +1,1334 @@
+        //             "l": 27781,         // Last tradeId
+        //             "T": 1498793709153, // Timestamp
+        //             "m": true,          // Was the buyer the maker?
+        //             "M": true           // Was the trade the best price match?
+        //         }
+        //     ]
+        //
+        // recent public trades and historical public trades
+        //
+        //     [
+        //         {
+        //             "id": 28457,
+        //             "price": "4.00000100",
+        //             "qty": "12.00000000",
+        //             "time": 1499865549590,
+        //             "isBuyerMaker": true,
+        //             "isBestMatch": true
+        //         }
+        //     ]
+        //
+        // const data = this.safeValue (response, []);
+        //
+        return this.parseTrades (response, market, since, limit);
+    }
+
+    parseOHLCV (ohlcv, market = undefined) {
+        return [
+            this.safeInteger (ohlcv, 0),
+            this.safeNumber (ohlcv, 1),
+            this.safeNumber (ohlcv, 2),
+            this.safeNumber (ohlcv, 3),
+            this.safeNumber (ohlcv, 4),
+            this.safeNumber (ohlcv, 5),
+        ];
+    }
+
+    async fetchOHLCV (symbol, timeframe = '1m', since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        // binancetr docs say that the default limit is 500, max=1500 for futures and max=1000 for spot markets
+        // the reality is that the time range wider than 500 candles doestn't work
+        const defaultLimit = 500;
+        const maxLimit = 1500;
+        const price = this.safeString (params, 'price');
+        params = this.omit (params, 'price');
+        limit = (limit === undefined) ? defaultLimit : Math.min (limit, maxLimit);
+        const request = {
+            'interval': this.timeframes[timeframe],
+            'limit': limit,
+        };
+        if (price === 'index') {
+            request['pair'] = market['id'];   // Index price takes this argument instead of symbol
+        } else {
+            // request['symbol'] = market['id'];
+            request['symbol'] = market['base'] + market['quote']; // Symbol needs to be without underscore for the endpoint
+        }
+        if (since !== undefined) {
+            request['startTime'] = since;
+            // if (market['inverse']) {
+            //     if (since > 0) {
+            //         const duration = this.parseTimeframe (timeframe);
+            //         const endTime = this.sum (since, limit * duration * 1000 - 1);
+            //         const now = this.milliseconds ();
+            //         request['endTime'] = Math.min (now, endTime);
+            //     }
+            // }
+        }
+        const method = 'publicGetKlines';
+        const response = await this[method] (this.extend (request, params));
```

Comment:
Let's replace the above two lines (and all similar occurrences) with just one line (we don't need the `method` variable here):

```JavaScript
        const response = await this.publicGetKlines (this.extend (request, params));
```